pkg/testsuite/e2e.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testsuite
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/exposure-notifications-verification-server/pkg/config"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+)
+
+// E2ESuite contains E2E test configs and other useful data.
+type E2ESuite struct {
+	cfg   *config.E2EConfig
+	db    *database.Database
+	realm *database.Realm
+
+	adminKey, deviceKey string
+}
+
+// NewAdminAPIClient returns an admin API client.
+func (s *E2ESuite) NewAdminAPIClient(context.Context, testing.TB) (*AdminClient, error) {
+	return NewAdminClient(s.cfg.AdminAPIURL, s.adminKey)
+}
+
+// NewAPIClient returns an API client.
+func (s *E2ESuite) NewAPIClient(context.Context, testing.TB) (*APIClient, error) {
+	return NewAPIClient(s.cfg.APIServerURL, s.deviceKey)
+}
+
+// NewE2ESuite returns an E2E test suite.
+func NewE2ESuite(tb testing.TB, ctx context.Context) *E2ESuite {
+	cfg := config.NewE2EConfig(tb, ctx)
+	db, err := cfg.DBConfig.Load(ctx)
+	if err != nil {
+		tb.Fatalf("failed to connect to database: %v", err)
+	}
+	if err := db.Open(ctx); err != nil {
```

Comment:
Do we want any retry here?

Reply:
db.Open has retry logic inside so I guess it's safe not to add another layer